SpliceAI.pm

```diff
@@ -268,11 +271,32 @@ sub run {
         $hash{'SpliceAI_cutoff'} = $result_flag;
       }
 
-      return ($self->{config}->{output_format} eq "json" || $self->{config}->{rest}) ?  {SpliceAI => \%hash} : \%hash;
+      $hash_aux{$data_value->{gene}} = \%hash;
+    }
+  }
+
+  return {} unless(%hash_aux);
+
+  my $result = {};
+
+  my $n_genes = scalar keys %hash_aux;
+  if($n_genes == 1) {
+    # Get the first and only gene from the hash of results
+    my $key_gene = (keys %hash_aux)[0];
+    $result = ($self->{config}->{output_format} eq "json" || $self->{config}->{rest}) ?  {SpliceAI => $hash_aux{$key_gene}} : $hash_aux{$key_gene};
+  }
+  elsif($n_genes > 1) {
+    # Compare genes from SpliceAI with the variant gene
+    my $gene_symbol = $tva->transcript->{_gene_symbol} || $tva->transcript->{_gene_hgnc};
+    if($hash_aux{$gene_symbol}) {
+      $result = ($self->{config}->{output_format} eq "json" || $self->{config}->{rest}) ?  {SpliceAI => $hash_aux{$gene_symbol}} : $hash_aux{$gene_symbol};
+    }
+    else {
+      warn("Warning: the variant gene symbol doesn't match any gene symbol from SpliceAI");
```

Comment:
The warning is good here! But for the case where $n_genes == 1 you are not checking if the key_gene matches the gene name of your overlapping transcript. So as suggest in my other comment I would store all data in $hash_aux and then get the results with $gene_symbol and if you don't get anything back but there is data in $hash_aux you can return the warning. I would also add some more information to the warning like the variant id, position, and gene_symbol.